src/plugins/gui/providers/revolutProvider.ts

```diff
@@ -0,0 +1,278 @@
+        const fiatKey = params.fiatCurrencyCode
+        const revolutFiat = asMaybe(asRevolutFiat)(supportedAssets.getFiatInfo(fiatKey))
+        const cryptoKey = `${params.pluginId}:${params.tokenId}`
+        const revolutCrypto = asMaybe(asRevolutCrypto)(supportedAssets.getCryptoInfo(cryptoKey))
+
+        if (revolutFiat == null) {
+          throw new Error(`Revolut fiat not found: ${fiatKey}`)
+        }
+        if (revolutCrypto == null) {
+          throw new Error(`Revolut crypto not found: ${cryptoKey}`)
+        }
+
+        const quoteData = await fetchRevolutQuote({
+          fiat: revolutFiat.currency,
+          amount: params.exchangeAmount,
+          crypto: revolutCrypto.id,
+          payment: 'revolut', // Only revolut is supported at the moment
+          region: params.regionCode.countryCode
+        })
+
+        const cryptoAmount = quoteData.crypto.amount.toString()
+        const fiatAmount = params.exchangeAmount
+
+        return {
+          providerId,
+          partnerIcon,
+          pluginDisplayName,
+          displayCurrencyCode: params.displayCurrencyCode,
+          cryptoAmount,
+          isEstimate: false,
+          fiatCurrencyCode: params.fiatCurrencyCode,
+          fiatAmount,
+          direction: params.direction,
+          expirationDate: undefined,
+          regionCode: params.regionCode,
+          paymentTypes: params.paymentTypes,
+
+          approveQuote: async (approveParams: FiatProviderApproveQuoteParams) => {
+            const { showUi, coreWallet } = approveParams
+            const walletAddresses = await coreWallet.getAddresses({ tokenId: params.tokenId })
+            const walletAddress = walletAddresses[0]?.publicAddress
+
+            if (walletAddress == null) {
+              throw new Error('No wallet address found')
+            }
+
+            const successReturnURL = encodeURIComponent('https://return.edge.app/fiatprovider/buy/revolut?transactionStatus=success')
+
+            const orderId = await io.makeUuid()
+
+            const { ramp_redirect_url: redirectUrl } = await fetchRevolutRedirectUrl({
+              fiat: revolutFiat.currency,
+              amount: parseFloat(fiatAmount),
+              crypto: quoteData.crypto.currencyId,
+              payment: 'revolut',
+              region: params.regionCode.countryCode,
+              wallet: walletAddress,
+              partnerRedirectUrl: successReturnURL,
+              orderId
+            })
+
+            console.log('!@! redirectUrl', redirectUrl)
+
+            await showUi.openExternalWebView({
+              url: redirectUrl,
+              providerId,
+              async deeplinkHandler(link) {
+                if (link.direction === 'sell') {
+                  throw new Error('Unexpected sell return link. Sell not supported by Revolut.')
+                }
+                const { transactionStatus } = link.query
+                if (transactionStatus === 'success') {
```

Comment:
What other `transactionStatus` values are there? Looks like we'll just do nothing if we get a different status. At minimum we need an error.